src/radical/pilot/utils/prof_utils.py

```diff
@@ -199,6 +199,114 @@
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.UMGR_STAGING_INPUT          }  ], # push to mongodb
+        'u_umngr_stage_in'             : [ {ru.EVENT: 'state'           , ru.STATE: rps.UMGR_STAGING_INPUT          }, 
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_STAGING_INPUT_PENDING }  ], # wait in mongodb
+        'u_agent_stage_in_queue'       : [ {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_STAGING_INPUT_PENDING },  
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_STAGING_INPUT         }  ], # pull from mongodb 
+        'u_agent_stage_in'             : [ {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_STAGING_INPUT         },  
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_SCHEDULING_PENDING    }  ], 
+        'u_agent_schedule_queue'       : [ {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_SCHEDULING_PENDING    },  
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_SCHEDULING            }  ],
+        'u_agent_schedule'             : [ {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_SCHEDULING            },  
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_EXECUTING_PENDING     }  ], 
+        'u_agent_execute_queue'        : [ {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_EXECUTING_PENDING     },  
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_EXECUTING             }  ], 
+        'u_agent_execute_prepare'      : [ {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_EXECUTING             },  
+                                           {ru.EVENT: 'exec_mkdir'      , ru.STATE: None                            }  ], 
+        'u_agent_execute_mkdir'        : [ {ru.EVENT: 'exec_mkdir'      , ru.STATE: None                            },
+                                           {ru.EVENT: 'exec_mkdir_done' , ru.STATE: None                            }  ],
+        'u_agent_execute_layer_start'  : [ {ru.EVENT: 'exec_mkdir_done' , ru.STATE: None                            },
+                                           {ru.EVENT: 'exec_start'      , ru.STATE: None                            }  ],
+        'u_agent_execute_layer'        : [ {ru.EVENT: 'exec_start'      , ru.STATE: None                            },
+                                          [{ru.EVENT: 'exec_ok'         , ru.STATE: None                            },
+                                           {ru.EVENT: 'exec_fail'       , ru.STATE: None                            } ]], # orte, ssh, mpi, ... 
+        'u_agent_cu_start'             : [ {ru.EVENT: 'cu_start'        , ru.STATE: None                            },
```

Comment:
the `_cu_` midfix (?) here refers to the unit shell script.  We may want to rename this (and likely should also rename the events), maybe to `_lm_` (for launch method, as those are somewhat launch method specific events), or `_script_` (as those are take in the unit shell script) - but the latter would be difficult to apply to function execution (`_lm_` would work there).

Reply:
'lm' is probably more difficult to parse than 'u' or 'p' but done as 'cu' was redundant with 'u'.